libs/hex-display-bytes/src/bytes.rs

```diff
@@ -0,0 +1,213 @@
+/// Adapted from ethers-core (https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/types/bytes.rs)
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    borrow::Borrow,
+    clone::Clone,
+    fmt::{Debug, Display, Formatter, LowerHex, Result as FmtResult},
+    ops::Deref,
+    str::FromStr,
+};
+use thiserror::Error;
+
+/// Wrapper type around Bytes to deserialize/serialize "0x" prefixed ethereum hex strings
+#[derive(Clone, Default, PartialEq, Eq, Hash, Serialize, Deserialize, Ord, PartialOrd)]
+pub struct Bytes(
+    #[serde(
+        serialize_with = "serialize_bytes",
+        deserialize_with = "deserialize_bytes"
+    )]
+    pub bytes::Bytes,
+);
+
+fn bytes_to_hex(b: &Bytes) -> String {
+    hex::encode(b.0.as_ref())
+}
+
+impl Debug for Bytes {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        write!(f, "Bytes(0x{})", bytes_to_hex(self))
```

Comment:
may be use formatter functions here? https://doc.rust-lang.org/std/fmt/trait.Debug.html#examples

something like `f.debug_tuple("Bytes").field(format!("0x{}", bytes_to_hex(self)))`

Reply:
Didn't get why it is better. Could you please clarify?

Reply:
In that case the final result would be `'Bytes("0xcafe..")'`, but we need to get `Bytes(0xcafe)` instead